### PR TITLE
Kashirin Alexander Lab4 Var 3

### PR DESCRIPTION
**Pull request by AlexanderVladislavovich**

### CODE REVIEW

File: mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  %1 = arith.constant 1 : i32
+  %2 = arith.subi %c5_i32, %1 : i32
+  %3 = arith.subi %c3_i32, %1 : i32
+  %4 = arith.cmpi sgt, %c5_i32, %0 : i32
+  scf.if %4 {
+    %5 = arith.cmpi sgt, %c3_i32, %0 : i32
+    scf.if %5 {
+      %6 = arith.subi %c5_i32, %1 : i32
+      %7 = arith.subi %3, %1 : i32
+    }
```
Comment (aobolensk) [inline mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir:47]: Add check for if with else branch

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir (hunk 1 above)
Comment (AlexanderVladislavovich) [reply mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir:47]: else branch added

File: mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// RUN: split-file %s %t
+// RUN: mlir-opt -load-pass-plugin=%mlir_lib_dir/KashirinMaxDepthPass%shlibext --pass-pipeline="builtin.module(func.func(KashirinMaxDepthPass))" %t/func1.mlir | FileCheck %t/func1.mlir
+// RUN: mlir-opt -load-pass-plugin=%mlir_lib_dir/KashirinMaxDepthPass%shlibext --pass-pipeline="builtin.module(func.func(KashirinMaxDepthPass))" %t/func2.mlir | FileCheck %t/func2.mlir
+// RUN: mlir-opt -load-pass-plugin=%mlir_lib_dir/KashirinMaxDepthPass%shlibext --pass-pipeline="builtin.module(func.func(KashirinMaxDepthPass))" %t/func3.mlir | FileCheck %t/func3.mlir
+
+//--- func1.mlir
+func.func @func1(%arg0: i32) -> i32 {
```
Comment (m-ly4) [inline mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir:8]: Please add also test case for `llvm.func` type.

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

File: mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir (hunk 2 above)
Comment (AlexanderVladislavovich) [reply mlir/test/Transforms/lab4/kashirin_alexander/tests.mlir:8]: llvm.func test case added

**Review by aobolensk — APPROVED** [review]

**Pull request merged**